tests/cli/test_buildspec.py

```diff
@@ -144,12 +144,15 @@ def test_buildspec_maintainers():
 @pytest.mark.cli
 def test_buildspec_find_invalid():
     cache = BuildspecCache(configuration=configuration)
-    cache.print_invalid_buildspecs(error=True)
-    cache.print_invalid_buildspecs(error=False)
-    cache.print_invalid_buildspecs(error=True, terse=True)
-    cache.print_invalid_buildspecs(error=False, terse=True)
-    cache.print_invalid_buildspecs(error=True, terse=True, header=True)
-    cache.print_invalid_buildspecs(error=False, terse=True, header=True)
+
+    # testing buildtest buildspec find invalid. This will assert SystemExit exception raised by sys.exit
+    with pytest.raises(SystemExit):
+        cache.print_invalid_buildspecs(error=True)
+        cache.print_invalid_buildspecs(error=False)
+        cache.print_invalid_buildspecs(error=True, terse=True)
+        cache.print_invalid_buildspecs(error=False, terse=True)
+        cache.print_invalid_buildspecs(error=True, terse=True, header=True)
+        cache.print_invalid_buildspecs(error=False, terse=True, header=True)
```

Comment:
i think for this you should invoke each call in its own context manager via `with` since if we decide to change the functionality of this method we want to make sure that the test works as expected. Right now this code will execute the first invocation raise an exception and we can't determine if all other calls will raise exception.

I think in future we will want to raise exception only if there is one or more entry that is invalid otherwise dont return an exit 1.

Reply:
Done !

Reply:
Done! @shahzebsiddiqui